### PR TITLE
Add fail cases to integration tests for Create organization

### PR DESCRIPTION
**Pull request by PiotrPrzybylak**
Changelog: None

Signed-off-by: Piotr Przybylak <piot.przybylak@gmail.com>

### CODE REVIEW

File: tests/tests/smtpd_mock.py [hunk 1]
```diff
@@ -21,5 +21,9 @@ def __init__(self, *args, **kwargs):
         self.received = False
         smtpd.SMTPServer.__init__(self, *args, **kwargs)
 
-    def process_message(self, *args, **kwargs):
+    def process_message(self, peer, mailfrom, rcpttos, data):
+        print(peer)
```
Comment (mchalski) [inline tests/tests/smtpd_mock.py]: I was thinking about asserting sth about the message, like title and len>0. but I'm ok with what we have here, so it's an 'approve'. we have all of this covered pretty well.

File: tests/tests/test_create_organization.py [hunk 2]
```diff
@@ -57,6 +56,26 @@ def test_success(self):
...
+        assert rsp.status_code == 409
+
+    @pytest.mark.usefixtures("multitenancy_setup_without_client_with_smtp")
+    def test_duplicate_email(self):
+        payload = {"request_id": "123456", "organization": "tenant-foo", "email":"some.user@example.com", "password": "asdfqwer1234", "g-recaptcha-response": "foobar"}
+        rsp = requests.post("https://%s/api/management/v1/tenantadm/tenants" % get_mender_gateway(), data=payload, verify=False)
+        assert rsp.status_code == 202
+        payload = {"request_id": "123457", "organization": "tenant-foo2", "email":"some.user@example.com", "password": "asdfqwer1234", "g-recaptcha-response": "foobar"}
+        rsp = requests.post("https://%s/api/management/v1/tenantadm/tenants" % get_mender_gateway(), data=payload, verify=False)
+        print(rsp.text)
```
Comment (GregorioDiStefano) [inline tests/tests/test_create_organization.py]: can you either remove this, or us logging?

File: tests/tests/test_create_organization.py [hunk 3]
```diff
@@ -42,13 +42,12 @@ def test_success(self):
...
         rsp = requests.post("https://%s/api/management/v1/tenantadm/tenants" % get_mender_gateway(), data=payload, verify=False)
 
         print("TestCreateOrganization: workflow started. Waiting...")
         for i in range(100):
-            print("waiting: ", i)
-            if smtp_mock.server.received == True:
+            if len(smtp_mock.server.messages) > 0:
                 break
             time.sleep(0.5)
         print("TestCreateOrganization: Waiting finished. Stoping mock")
```
Comment (GregorioDiStefano) [inline tests/tests/test_create_organization.py]: remove or use logging

File: tests/tests/test_create_organization.py [hunk 4]
```diff
@@ -42,13 +42,12 @@ def test_success(self):
 
         print("TestCreateOrganization: making request")
 
-        payload = {"request_id": "123456", "tenant_id":"123456", "organization": "tenant-foo", "email":"some.user@example.com", "password": "asdfqwer1234", "g-recaptcha-response": "foobar"}
+        payload = {"request_id": "123456", "organization": "tenant-foo", "email":"some.user@example.com", "password": "asdfqwer1234", "g-recaptcha-response": "foobar"}
         rsp = requests.post("https://%s/api/management/v1/tenantadm/tenants" % get_mender_gateway(), data=payload, verify=False)
 
         print("TestCreateOrganization: workflow started. Waiting...")
```
Comment (GregorioDiStefano) [inline tests/tests/test_create_organization.py]: remove or use logging

**Pull request merged**